cli/src/main/java/com/devonfw/tools/ide/tool/IdeasyCommandlet.java

```diff
@@ -228,71 +230,13 @@ private void installIdeasyWindowsEnv(Path ideRoot, Path installationPath) {
   private void setGitLongpaths() {
     GitContext gitContext = new GitContextImpl(this.context);
     gitContext.verifyGitInstalled();
-    setGitConfigProperty("core", "longpaths", "true", this.context.getUserHome().resolve(".gitconfig"));
-  }
-
-  /**
-   * @param section the section to modify
-   * @param property the property to set
-   * @param value the value to set
-   * @param configPath the path of the config file
-   */
-  public void setGitConfigProperty(String section, String property, String value, Path configPath) {
-    String gitconfig;
-    // read files
-    FileAccess fileAccess = this.context.getFileAccess();
-    gitconfig = fileAccess.readFileContent(configPath);
-    if (gitconfig == null) {
-      gitconfig = "";
-    }
-
-    LinkedHashMap<String, LinkedHashMap<String, String>> iniMap = parseIniFile(gitconfig);
-
-    // check if section exists
-    if (!iniMap.containsKey(section)) {
-      iniMap.put(section, new LinkedHashMap<>());
-    }
-
-    // set property
-    iniMap.get(section).put(property, value);
-
-    // write out new file
-    StringBuilder newConfig = new StringBuilder();
-    for (String configSection : iniMap.keySet()) {
-      newConfig.append(String.format("[%s]\n", configSection));
-      LinkedHashMap<String, String> properties = iniMap.get(configSection);
-      for (String sectionProperty : properties.keySet()) {
-        String propertyValue = properties.get(sectionProperty);
-        newConfig.append(String.format("\t%s = %s\n", sectionProperty, propertyValue));
-      }
-    }
-
-    try {
-      Files.writeString(configPath, newConfig.toString());
-    } catch (IOException e) {
-      this.context.error("could not write git config file at %s", configPath);
-    }
-  }
-
-  private LinkedHashMap<String, LinkedHashMap<String, String>> parseIniFile(String iniFile) {
-    List<String> iniLines = iniFile.lines().toList();
-    LinkedHashMap<String, LinkedHashMap<String, String>> iniMap = new LinkedHashMap<>();
-    String currentSection = "";
-    for (String line : iniLines) {
-      if (line.isEmpty()) {
-        continue;
-      }
-      if (line.startsWith("[")) {
-        currentSection = line.replace("[", "").replace("]", "");
-        iniMap.put(currentSection, new LinkedHashMap<>());
-      } else {
-        String[] parts = line.split("=");
-        String propertyName = parts[0].trim();
-        String propertyValue = parts[1].trim();
-        iniMap.get(currentSection).put(propertyName, propertyValue);
-      }
-    }
-    return iniMap;
+    Path configPath = this.context.getUserHome().resolve(".gitconfig");
+    IniFile iniFile = new IniFileImpl();
+    FileAccess fileAccess = new FileAccessImpl(context);
+    fileAccess.readIniFile(configPath, iniFile);
```

Comment:
Do not create a new FileAccessImpl but use the existing one. Also you shall not know about the implementation here - we might even have a different implementation in tests, who knows (implementation secret pattern). Also you created a nice convenience method to create the `iniFile` automatically that you can use to simplify to a single line:
```suggestion
    IniFile iniFile = this.context.getFileAccess().readIniFile(configPath);
```